### PR TITLE
3009 sos imagery

### LINKED ISSUE #3239: Loading share data doesn't update time of a currently added layer
Can be reproduced with 2 share URLs. Navigate the the first, then copy and paste the second into the URL bar to go to the second:
- http://ci.terria.io/master/#share=s-80TUGBkAJdzKKGrZKjvaysjJb5v
- http://ci.terria.io/master/#share=s-rJgPc0qmUDWMR3DeKxx93boIb6h

Note the date doesn't change to 19/02/2019 even though the second share link goes to that date when freshly loaded.

### PR DESCRIPTION
**Pull request by stevage**
This implements the first part of #3009. It has some limitations:

- doesn't support multiple imagery layers
- has its own way of dealing with clicking on the chart, to try to find the right imagery for a given time. Which sort of duplicates existing code.

I tried to fix the second problem by integrating with existing the existing concept of `.intervals` in the 3009-sos-imagery-simplified branch (https://github.com/TerriaJS/terriajs/tree/3009-sos-imagery-simplified) but it got hard, and seems to be more effort than it's worth at this stage.

---
Also fixes https://github.com/TerriaJS/terriajs/issues/3239

### CODE REVIEW

**Comment by kring** [comment]
Something weird going on with the axes here:
image

**Review by kring — COMMENTED** [review]
Here's some trivial comments from a review I did awhile ago that I apparently forgot to submit.

The real stuff is:
* The weird axes rendering in my screenshot above.
* Remove the WIP data series checkbox thing below the chart and related code, that should be a separate PR.
* I'm a little concerned about the the pretty intrusive and bespoke changes to `ImageryLayerCatalogItem`. And I think part of it comes back to a UX question: will users understand that clicking a point on the map will make _some_ layers display their availability at that particular point on the chart? Even though once that feature info is closed, there's nothing in the UI to indicate that you're looking at availability at that point? Would it make sense for it to be a more explicit user action (i.e. "add availability of this layer at this point to the chart")?

File: lib/Charts/ChartData.js [hunk 1]
```diff
@@ -72,6 +73,25 @@ var ChartData = function(points, parameters) {
...
+     * Chart type. If you want these points to be rendered with a certain way. Leave empty for auto detection.
+     * @type {String}
+     */
+    this.type = parameters.type;
+
+    /**
+     * Click handler (called with (x, y) in data units) if some special behaviour is required on clicking.
+     * @type {Function}
+     */
+    this.onClick = parameters.onClick;
```
[the remark this answers is not in the document]
Comment (stevage) [reply lib/Charts/ChartData.js:89]: Done. Seems odd that each of these properties has to be documented twice.

**Comment by stevage** [comment]
@kring your thoughts on the above?

**Comment by kring** [comment]
I don't know the answer @stevage, but the goal makes sense, right? Keep the chart UI changes separate from this feature. I don't know if you'll be able to do this magically with git, it sounds like maybe not.

If you feel it's too hard and/or doesn't makes sense to do, then that means this PR is gated on _finishing_ the chart UI overhaul. I'm not necessarily opposed to that (especially if Phil likes the idea), but it's not going to get this merged faster, that's for sure.

**Comment by kring** [comment]
To do:
* [x] Do we need a "Show on chart" button the workbench, next to the date/time picker? Currently it's configurable in the catalog but not through the UI. The Invision shows such a button, but it appears to have replaced the "Use timeline" button. I think we need both.
* [x] Availability filtering doesn't survive sharing.
* [x] Separate from but related to the above, the currentTime seems to get lost completely upon sharing in this branch.
* Probably more?

**Comment by philipgrimmett** [comment]
@kring [timeline-chart-icons.zip](https://github.com/TerriaJS/terriajs/files/2887105/timeline-chart-icons.zip)

Updated invasion for date picker changes:
https://invis.io/MTQMK8VER6P#/347826475_datepicker-1

**Comment by kring** [comment]
From our meeting today:
* [ ] There are two obviously bad URLs in the `stationIdWhitelist` in the `neii-demo.json` init file.
* [ ] If you add a satellite imagery layer after the feature info is already open, the "show at time" button for that layer does not appear.
* [x] Something made the chart go crazy, possibly related to clicking the Expand on a chart button after adding a satellite imagery layer with the feature info already open.

**Comment by meh9** [comment]
@kring it appears that chart y axis scale has disappeared?

Screen Shot 2019-03-13 at 15 04 17

**Comment by kring** [comment]
What dataset is that @meh9?

**Comment by AnaBelgun** [comment]
@kring that's from DroughtMap: https://map.drought.gov.au/#share=s-hfoJfwLovGqw9QesbjPvL0iM2To 
he probably dragged it: https://droughtreliefpublic.blob.core.windows.net/nationalmapdatapublic/Farm-Businesses/FMD%20industry.csv

**Pull request merged**